src/features/Details/containers/Details.tsx

```diff
@@ -0,0 +1,80 @@
+import { useContext, useEffect } from 'react';
+import { useNavigate, useParams } from 'react-router-dom';
+
+import { AppContext } from '@context/AppContext';
+import { NotFound } from '@components/NotFound/NotFound';
+import { ReactComponent as RippleLoader } from '@svg/ripple.svg';
+import { jestIdsMap } from '@utils/jestHelpers';
+
+import { HeroHeadline } from '@features/Details/components/HeroHeadline/HeroHeadline';
+import { Powerstats } from '@features/Details/components/Powerstats/Powerstats';
+import { Biography } from '@features/Details/components/Biography/Biography';
+import { Appearance } from '@features/Details/components/Appearance/Appearance';
+import { Work } from '@features/Details/components/Work/Work';
+import { Connections } from '@features/Details/components/Connections/Connections';
+import { Image } from '@features/Details/components/Image/Image';
+
+import { Button } from '@components/Button/Button';
+import { withContext } from '@components/withContext/WithContext';
+import { DetailsContext, DetailsProvider } from '../context/DetailsContext';
+
+import styles from './Details.module.scss';
+
+export const Details = () => {
+  const { slug } = useParams();
+  const { areHeroesFetching } = useContext(AppContext);
+  const { areDetailsLoading, superhero, fetchDetails } =
+    useContext(DetailsContext);
```

Comment:
**question** why do you need to know here if heroes are fetching?


**thought** apart from the fact that the backend provides you with the heroes/{id} endpoint, I guess you wanted to reuse previously fetched data, which is great, but it makes this code less predictable and understandable since you're using AppContext in DetailsContext anyway

Reply:
> why do you need to know here if heroes are fetching?

@kamil-borecki-cc It's because I need to load all heroes before I can display any details. So the Details view displays a loader when `areHeroesFetching` ("true" fetching) or `areDetailsLoading` (a fake delay) is true.

I agree with your thought though, and I only let myself do that in this specific case where I have the entire data in my browser's memory